compiler/rustc_ast_lowering/src/errors.rs

```diff
@@ -1,10 +1,10 @@
 use rustc_errors::{
     codes::*, AddToDiag, Diag, DiagArgFromDisplay, EmissionGuarantee, SubdiagMessageOp,
 };
-use rustc_macros::{Diagnostic, Subdiagnostic};
+use rustc_macros::{AddToDiag, IntoDiag};
 use rustc_span::{symbol::Ident, Span, Symbol};
 
-#[derive(Diagnostic)]
+#[derive(IntoDiag)]
```

Comment:
Does it make sense to go this way or the opposite and rename the traits? `#[derive(Diag)]` and `trait Diag` might be better than `#[derive(IntoDiag)]` and `trait IntoDiag`?

Reply:
Hmm, good suggestion. We'd have `derive(Diag)`, `derive(Subdiag)`, `derive(LintDiag)`. I'll do that.

Reply:
Turns out that doesn't work because `Diag` and `Subdiag` are already names of types, so they can't also be names of traits. Instead I'll revert to the original `derive(Diagnostic)`, etc., and change the trait names to match.